### PR TITLE
Lobby websocket

### LINKED ISSUE #6: Add lobby websocket

### PR DESCRIPTION
**Pull request by Blindeen**

### CODE REVIEW

File: plemiona_api/management.py [hunk 1]
```diff
@@ -60,3 +59,52 @@ def create_jwt(cls, user):
...
+
+from channels.db import database_sync_to_async
+from rest_framework_simplejwt.exceptions import InvalidToken, TokenError
+from rest_framework_simplejwt.tokens import UntypedToken
+from channels.middleware import BaseMiddleware
+from channels.auth import AuthMiddlewareStack
+from django.db import close_old_connections
+from urllib.parse import parse_qs
+from jwt import decode as jwt_decode
+from django.conf import settings
```
Comment (MendelDamian) [inline plemiona_api/management.py]: Move the imports to the top of the file + reorder them (first goes inbuilt packages, then 3rd party, then from project)

File: game/consumers.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+from game.models import Player
+from game.serializers import PlayerSerializer
+
+class LobbyConsumer(WebsocketConsumer):
+    def connect(self):
+        player = self.scope.get("player", None)
+        if not player:
+            self.close()
+
```
Comment (MendelDamian) [inline game/consumers.py]:
```suggestion
        if not player:
            self.close()
            return

```

File: game/consumers.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import json
+from channels.generic.websocket import WebsocketConsumer
+from asgiref.sync import async_to_sync
+
+from game.models import Player
+from game.serializers import PlayerSerializer
+
+class LobbyConsumer(WebsocketConsumer):
```
Comment (MendelDamian) [inline game/consumers.py]: You can try to re-write the consumer making it asynchronous. It's not important, just if you want
[https://channels.readthedocs.io/en/stable/tutorial/part_3.html](https://channels.readthedocs.io/en/stable/tutorial/part_3.html)

**Review by MendelDamian — DISMISSED** [review]
Good Job

**Review by Vittoriio — DISMISSED** [review]
Nice data structure

File: plemiona_api/management.py [hunk 4]
```diff
@@ -60,3 +67,46 @@ def create_jwt(cls, user):
...
+        # Try to authenticate the user
+        try:
+            # This will automatically validate the token and raise an error if token is invalid
+            UntypedToken(token)
+        except (InvalidToken, TokenError) as e:
+            # Token is invalid
+            return await self.inner(scope, receive, send)
+        else:
+            #  Then token is valid, decode it
+            decoded_data = jwt_decode(token, settings.SECRET_KEY, algorithms=["HS256"])
```
Comment (MendelDamian) [inline plemiona_api/management.py]: You can use `jwt.deocde` here and remove importing this
```suggestion
            decoded_data = jwt.decode(token, settings.SECRET_KEY, algorithms=["HS256"])
```

**Review by MendelDamian — DISMISSED** [review]
Rock solid 🚀

**Review by MendelDamian — APPROVED** [review]

**Review by Vittoriio — APPROVED** [review]
Cool 🗿

**Pull request merged**